pkg/catalog/manager/crd.go

```diff
@@ -12,14 +13,28 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 const (
 	CatalogNameLabel  = "catalog.cattle.io/name"
 	TemplateNameLabel = "catalog.cattle.io/template_name"
 )
 
+type InvalidChartError struct {
+	Err          string
+	TemplateName string
+}
+
+func (e *InvalidChartError) Error() string {
+	return fmt.Sprintf("failed %v %s", e.TemplateName, e.Err)
+}
+
 func (m *Manager) createTemplate(template v3.CatalogTemplate, catalog *v3.Catalog) error {
+	// check to see if the template has valid urls - do not process if none exist
+	if len(template.Spec.Versions) == 0 || template.Spec.Versions == nil {
+		return &InvalidChartError{"must have one or more valid urls", template.Name}
```

Comment:
The error doesn't convey what really happened, there isn't any data, not that it's invalid.

Reply:
removed as part of new implementation